pandas/core/indexes/interval.py

```diff
@@ -461,7 +461,28 @@ def is_unique(self):
         """
         Return True if the IntervalIndex contains unique elements, else False
         """
-        return self._multiindex.is_unique
+        left = self.values.left
+        right = self.values.right
+
+        def _is_unique(left, right):
```

Comment:
If my previous comment is correct, I don't think we need this to be a function anymore since it's only called once, so you can just put the function's logic at the end of the method.

Can you also test out the following variant of `_is_unique`:
```python
from collections import defaultdict


def _is_unique2(left, right):
    seen_pairs = defaultdict(bool)
    check_idx = np.where(left.duplicated(keep=False))[0]
    for idx in check_idx:
        pair = (left[idx], right[idx])
        if seen_pairs[pair]:
            return False
        seen_pairs[pair] = True

    return True
```
I did a sample run of this, and it appears to be a bit more efficient:
```python
In [3]: np.random.seed(123)
   ...: left = pd.Index(np.random.randint(5, size=10**5))
   ...: right = pd.Index(np.random.randint(10**5/4, size=10**5))

In [4]: %timeit _is_unique(left, right)
3.84 ms ± 34.7 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [5]: %timeit _is_unique2(left, right)
1.13 ms ± 26.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```
I haven't fully tested this in all scenarios though.

Reply:
HEAD adopts `_is_unique2` and HEAD~3 adopts `_is_unique`. The performance is slightly worse but the code is more explanatory. 

```
       before           after         ratio
     [4ec1fe97]       [202b2cfa]
     <intv-is-unique~3>       <intv-is-unique>
        230±0.2ns          217±2ns     0.94  index_object.IntervalIndexMethod.time_is_unique(1000)
+         277±2ns          316±5ns     1.14  index_object.IntervalIndexMethod.time_is_unique(100000)
```

Reply:
are these asv's really short? maybe have a longer one and see how this scales

Reply:
Yeah, I was wondering this too; `is_unique` is cached, so I wonder if the asv is just timing the cache lookup?  Does anything special need to be done to handle things that are cached?

Reply:
HEAD~6 adopts `_is_unique` while HEAD adopts `_is_unique2`.

```
       before           after         ratio
     [4ec1fe97]       [d3af9c91]
     <intv-is-unique~6>       <intv-is-unique>
          223±1ns          207±1ns     0.93  index_object.IntervalIndexMethod.time_is_unique(1000)
+         270±5ns          302±4ns     1.12  index_object.IntervalIndexMethod.time_is_unique(100000)
       1.50±0.01s          1.84±0s    ~1.22  index_object.IntervalIndexMethod.time_is_unique(10000000)
```